lib/atlas/scaler/area_attributes_scaler.rb

```diff
@@ -0,0 +1,27 @@
+module Atlas
+  class Scaler::AreaAttributesScaler
+    # Only attributes common to FullDataset and DerivedDataset
+    # may be scaled
+    SCALEABLE_AREA_ATTRS = Atlas::Dataset.attribute_set
+      .select { |attr| attr.options[:proportional] }.map(&:name).freeze
+
+    def self.call(*args)
+      new(*args).scale
+    end
+
+    def initialize(base_dataset, scaling_factor)
+      @base_dataset   = base_dataset
+      @scaling_factor = scaling_factor
+    end
+
+    def scale
+      result = {}
+      SCALEABLE_AREA_ATTRS.map do |attr|
+        if value = @base_dataset[attr]
+          result[attr] = Util::round_computation_errors(value * @scaling_factor)
+        end
+      end
+      result
```

Comment:
Personally, I still do not really like each_with_object. My version here looks more readable to me (except map instead each, that is a mistake of course). But I am fine either way